maestro/src/workflow.py

```diff
@@ -104,8 +105,17 @@ def to_mermaid(self, kind="sequenceDiagram", orientation="TD") -> str:
 
     async def run(self):
         """Execute workflow."""
-        self.create_or_restore_agents(self.agent_defs, self.workflow)
-        return await self._condition()
+        try:
+            self.create_or_restore_agents(self.agent_defs, self.workflow)
+            return await self._condition()
+        except Exception as err:
+            if self.workflow["spec"]["template"].get("exception"):
```

Comment:
do we need to explicitly handle the case where there is not exception in the template?

Reply:
I'm not sure.  How about re-throwing the exception?

Reply:
I think that works :+1: